client/state/form/actions.js

```diff
@@ -7,3 +14,68 @@ export const setFormProperty = ( field, value ) => {
 		value,
 	};
 };
+
+export const goToStep = ( stepIndex ) => {
+	return {
+		type: GO_TO_STEP,
+		step: stepIndex,
+	};
+};
+
+export const nextStep = () => ( dispatch, getState, { callbackURL, nonce, formSchema, formLayout } ) => {
```

Comment:
See `react-thunk` documentation: https://github.com/gaearon/redux-thunk

With `react-thunk`, actions can either be plain objects (such as the value returned from `gotoStep(stepIndex)` ), or functions that receive a way to dispatch more actions and access state directly. It's a bit hard to grasp at first.

So the `nextStep` action creator yields a function that, when invoked with `dispatch`, `getState` and a bunch of arguments (configured in `main.js` line 22), it executes all that logic. That logic can fully read the state (using the `getState` arg) and can dispatch other actions (using the `dispatch` arg).

This allows to have complex logic inside an action creator, even async logic such as a network request.